smack-extensions/src/main/java/org/jivesoftware/smackx/filetransfer/IBBTransferNegotiator.java

```diff
@@ -114,7 +120,42 @@ InputStream negotiateIncomingStream(Stanza streamInitiation) throws NotConnected
         private ByteStreamRequest(InBandBytestreamManager manager, Open byteStreamRequest) {
             super(manager, byteStreamRequest);
         }
+    }
+
+    InputStream getIbbIncomingStream(final StreamInitiation initiation) throws  InterruptedException, SmackException  {
 
+        IbbBytestreamListener listener = new IbbBytestreamListener();
+        manager.removeIgnoredBytestreamRequest(initiation.getSessionID());
+        manager.addIncomingBytestreamListener(listener);
+        return listener.getStream();
     }
 
+    private static class IbbBytestreamListener extends InBandBytestreamListener {
+
+        InputStream inputStream = null;
+
+        public InputStream getStream() throws InterruptedException, SmackException {
+
+            int wait = 50; // wait for 10 seconds
+            do {
+                Thread.sleep(200);
```

Comment:
> Please use a proper synchronization/notification mechanism to wait.

Actually I do not like this part of the code. The original implementation does not provide a convenience callback on timeout. For aTalk implementation, I just did a dirty fix to the problem. I do not have full knowledge to made substantial changes to the sources.

Below is aTalk UI implementation for the fault-tolerance file transfer.
[aTalk Fault-Tolerance File Sharing Implementation & Enhancements](https://atalk.sytes.net/atalk/faq.html#feature_05)

From the example, it can see that the fault-tolerance feature actually helps to simplify the whole UI in file sharing. Otherwise a layman may not know what to do if the file sending failed, even with user selectable options for all the sending file methods provided.